docs/rfcs/component-universal-telemetry.md

```diff
@@ -0,0 +1,212 @@
+# Pipeline Component Telemetry
+
+## Motivation and Scope
+
+The collector should be observable and this must naturally include observability of its pipeline components. Pipeline components
+are those components of the collector which directly interact with data, specifically receivers, processors, exporters, and connectors.
+
+It is understood that each _type_ (`filelog`, `batch`, etc) of component may emit telemetry describing its internal workings,
+and that these internally derived signals may vary greatly based on the concerns and maturity of each component. Naturally
+though, there is much we can do to normalize the telemetry emitted from and about pipeline components.
+
+Two major challenges in pursuit of broadly normalized telemetry are (1) consistent attributes, and (2) automatic capture.
+
+This RFC represents an evolving consensus about the desired end state of component telemetry. It does _not_ claim
+to describe the final state of all component telemetry, but rather seeks to document some specific aspects. It proposes a set of
+attributes which are both necessary and sufficient to identify components and their instances. It also articulates one specific
+mechanism by which some telemetry can be automatically captured. Finally, it describes some specific metrics and logs which should
+be automatically captured for each kind of pipeline component.
+
+## Goals
+
+1. Define attributes that are (A) specific enough to describe individual component[_instances_](https://github.com/open-telemetry/opentelemetry-collector/issues/10534)
+   and (B) consistent enough for correlation across signals.
+2. Articulate a mechanism which enables us to _automatically_ capture telemetry from _all pipeline components_.
+3. Define specific metrics for each kind of pipeline component.
+4. Define specific logs for all kinds of pipeline component.
+
+## Attributes
+
+All signals should use the following attributes:
+
+### Receivers
+
+- `otel.component.kind`: `receiver`
```

Comment:
OTel SemConv guidelines require metric attributes to be namespaces so they can be placed in a global registry along with resource attributes. I don't think we can use `otel` for collector. It probably should be `otelcol`. Or `otel.collector` instead. However, `otel.` seems like [kind of a restricted namespace](https://github.com/open-telemetry/semantic-conventions/blob/de8f94ed5c8601f6ef8d596906d4eb5c2ff895ec/docs/general/attribute-naming.md#otel-namespace). Not sure if we can use it or not. I think we need to run this by the otel semantic conventions WG group before merging the RFC.

Reply:
@dmitryax if you think contacting the semantic convention SIG is a requirement for this RFC to be merged, please request changes on the PR so that we follow [the process](https://github.com/open-telemetry/opentelemetry-collector/tree/main/docs/rfcs#merge-rules)

Reply:
I'm fine with changing the attribute names but delaying this functionality further for an entirely separate naming process seems unnecessary. Can we really not choose our own attributes names?

Reply:
This sentence from the doc linked by @dmitryax makes me hesitant about keeping the current approach:

> Any additions to the otel.* namespace MUST be approved as part of OpenTelemetry specification.

Reply:
> This sentence from the doc linked by @dmitryax makes me hesitant about keeping the current approach:
> 
> > Any additions to the otel.* namespace MUST be approved as part of OpenTelemetry specification.

I agree we can't just use the `otel` namespace. Can't we choose another on our own though? I like the `otelcol` suggestion.

Reply:
`otelcol` works for me, it's also the one we use for metrics so it makes sense to be consistent here

Reply:
@mx-psi asked me to chime in to help interpret the spec. 

You are reading it correctly. If you want to use `otel.` prefix it needs to go through semconv group.

What can you use instead? If you read the [preceding paragraph](https://github.com/open-telemetry/semantic-conventions/blob/de8f94ed5c8601f6ef8d596906d4eb5c2ff895ec/docs/general/attribute-naming.md#recommendations-for-application-developers) the closest applicable rules seems to be this:

>The name is specific to your company and may be possibly used outside the company as well. To avoid clashes with names introduced by other companies (in a distributed system that uses applications from multiple vendors) it is recommended to prefix the new name by your company's reverse domain name, e.g. com.acme.shopname.

If we consider "OpenTelemetry" the "company" then we should use "io.opentelemetry." as the org-specific prefix, e.g. "io.opentelemetry.collector." for the Collector. 

However, I am not sure I like such a long prefix. "otelcol" or "otel.collector" looks much nicer to me.

I cannot find spec recommendations about when a new top-level namespace (like "otelcol") is allowed to be used. I would argue the spec should have a rule like this and the rule should be that it needs to go through semconv, just like for anything under "otel." namespace.

My recommendation is this:
- Decide which prefix you want to use: "otelcol." or "otel.collector." appear to be your top choices.
- Submit a request to get this added to semconv. We should be able to convince semconv group to accept it and I do not anticipate difficulties here.

You don't need to be blocked or wait until acceptance by semconv group, there is a ton of work that needs to happen anyway and replacing metric names should be trivial anytime before you declare them stable.

One additional thing that I think is worth thinking about: is this Collector-only telemetry or it can be applicable to SDK processors/exporters? Collector's telemetry is going to be much richer and needs to be more expressive, so I am not sure if it is beneficial to model Collector and SDK the same way. However, if we end up deciding that indeed we want to model them the same way, with SDK's use cases being virtually a subset of Collector's use cases, then "otel." prefix seems to be the most natural choice.

Reply:
Thanks @tigrannajaryan. I'll open an issue to add `otelcol` as a top level namespace in the semantic conventions.

> One additional thing that I think is worth thinking about: is this Collector-only telemetry or it can be applicable to SDK processors/exporters? Collector's telemetry is going to be much richer and needs to be more expressive, so I am not sure if it is beneficial to model Collector and SDK the same way. However, if we end up deciding that indeed we want to model them the same way, with SDK's use cases being virtually a subset of Collector's use cases, then "otel." prefix seems to be the most natural choice.

This is intended to be collector-only telemetry. It is attributed to the collector's _instances_ of components, which are governed by a model that is not intended to be aligned with other parts of the project.